### PR TITLE
check is_inside_git_work_tree as part of gleam new

### PR DESCRIPTION
**Pull request by bookshelfdave**
addresses part of #1875, new CLI options could be in a separate PR.

### CODE REVIEW

File: compiler-cli/src/fs.rs [hunk 1]
```diff
@@ -549,9 +549,64 @@ pub fn hardlink(from: impl AsRef<Path> + Debug, to: impl AsRef<Path> + Debug) ->
...
+    assert!(is_inside_git_work_tree(Path::new(".")).unwrap());
+    assert!(!is_inside_git_work_tree(Path::new("/")).unwrap())
+}
+
 pub fn git_init(path: &Path) -> Result<(), Error> {
     tracing::debug!(path=?path, "initializing git");
 
+    if is_inside_git_work_tree(&path)? {
+        tracing::debug!(path=?path, "already inside a git work tree");
+        return Err(Error::GitInitialization {
```
Comment (lpil) [inline compiler-cli/src/fs.rs:605]: If we are in a git repo we want to not create a new one, not error

File: compiler-cli/src/fs.rs [hunk 2]
```diff
@@ -549,9 +549,64 @@ pub fn hardlink(from: impl AsRef<Path> + Debug, to: impl AsRef<Path> + Debug) ->
...
+fn is_inside_git_work_tree_test() {
+    assert!(is_inside_git_work_tree(Path::new(".")).unwrap());
+    assert!(!is_inside_git_work_tree(Path::new("/")).unwrap())
+}
+
 pub fn git_init(path: &Path) -> Result<(), Error> {
     tracing::debug!(path=?path, "initializing git");
 
+    if is_inside_git_work_tree(&path)? {
+        tracing::debug!(path=?path, "already inside a git work tree");
```
Comment (lpil) [inline compiler-cli/src/fs.rs:604]: The convention we use for event names is to have underscores.

```suggestion
        tracing::debug!(path=?path, "git_repo_already_exists");
```

**Review by lpil — COMMENTED** [review]
Nice, thank you!

Would you mind also updating the changelog file?

**Comment by lpil** [comment]
Let's just have the git detection in this PR, the additional flags etc can be a new PR.

**Comment by bookshelfdave** [comment]
will do, I just pushed a second commit up with the cli flags, but I'll apply that in a different PR. Thanks!

File: compiler-cli/src/fs.rs [hunk 3]
```diff
@@ -549,9 +549,62 @@ pub fn hardlink(from: impl AsRef<Path> + Debug, to: impl AsRef<Path> + Debug) ->
         .map(|_| ())
 }
 
+pub fn is_inside_git_work_tree(path: &Path) -> Result<bool, Error> {
+    tracing::debug!(path=?path, "check if we're inside a git work tree");
```
Comment (lpil) [inline compiler-cli/src/fs.rs:553]:
```suggestion
    tracing::debug!(path=?path, "checking_for_git_repo");
```

**Comment by bookshelfdave** [comment]
I'm assuming that I'll create an `Unreleased` section in CHANGELOG.md with my updates

**Review by lpil — APPROVED** [review]
Wonderful, thank you!

**Pull request merged**